src/core/Network.ts

```diff
@@ -35,26 +35,26 @@ export abstract class Network {
     public createAddressFromPublicKey(publicKey: Key): RawAddress {
         const publicKeyBytes = publicKey.toBytes();
         // step 1: sha3 hash of the public key
-        const publicKeyHash = this.addressHasher().update(Buffer.from(publicKeyBytes));
+        const publicKeyHash = this.addressHasher().arrayBuffer(publicKeyBytes);
 
         // step 2: ripemd160 hash of (1)
-        const ripemd160Hash = new ripemd160().update(publicKeyHash.digest()).digest();
+        const ripemd160Hash = new ripemd160().update(Buffer.from(publicKeyHash)).digest();
```

Comment:
same comment about `Buffer.from`. you should probably review all usages in this PR and/or project.

Reply:
Tried to use uin8array / dataview but it generate different hash?

Reply:
i checked the ripemd impl, it is quite old and doesn't appear to support native arrays (only buffers and strings). 
https://github.com/crypto-browserify/hash-base/blob/master/index.js#L50

anyway, it's ok since publicKeyHash is an `ArrayBuffer`, so Buffer shouldn't allocate any memory. 

can you remind me if this code works / how it is translated in a browser?

Reply:
Yup. it works in browsers. We don't have issues in the current sdk